feat: add environment variable support for all configuration options

ota2000 opened the pull request:
This PR adds environment variable support for all configuration options in fake-gcs-server. This is particularly useful when running the server in containerized environments like GitHub Actions service containers.

## Changes

- Added environment variable support for all configuration flags
- Added a generic helper function `envVarOrDefaultT` to handle type conversion
- Environment variables follow the pattern `FAKE_GCS_*` (e.g., `FAKE_GCS_PORT` for `-port`)
- Default values are used when environment variables are not set or invalid
- Maintains backward compatibility with command-line flags

## Related Issues/PRs

- Continues the work from #1608
- Addresses the need for easier configuration in containerized environments

## Testing

- All existing tests pass
- Added test coverage for environment variable handling
- Manually tested in containerized environment

This change makes it easier to configure fake-gcs-server in containerized environments while maintaining all existing functionality.

ota2000 commented:
@fsouza 

I've created this PR to add environment variable support for all configuration options. This enhancement makes it easier to configure fake-gcs-server in containerized environments, particularly in GitHub Actions service containers.

The implementation:
- Uses the `envVarOrDefaultT` helper function
- Maintains backward compatibility with command-line flags
- Includes comprehensive test coverage
- Follows the established pattern of `FAKE_GCS_*` for environment variable names

I would greatly appreciate your review when you have a chance. Thank you for maintaining this valuable project!

ota2000 commented:
@fsouza Fixed a failed test in CI.

fsouza commented on internal/config/config.go:
```diff
@@ -68,26 +81,69 @@ func Load(args []string) (Config, error) {
-	fs.StringVar(&logLevel, "log-level", "info", "level for logging. Options same as for logrus: trace, debug, info, warn, error, fatal, and panic")
+	fs.StringVar(&cfg.backend, "backend", envVarOrDefault("FAKE_GCS_BACKEND", filesystemBackend, func(s string) (string, error) {
+		return s, nil
+	}), "storage backend (memory or filesystem)")
```
can you extract the anonymous converter function as something like:

```go
func nopConverter(s string) (string, error) {
    return s, nil
}
```

fsouza commented on internal/config/config.go:
```diff
@@ -60,6 +62,17 @@ type EventConfig struct {
+	if val, ok := os.LookupEnv(key); ok && val != "" {
+		if converted, err := convert(val); err == nil {
+			return converted
+		}
```
I feel like if `convert` returns an error we should propagate it. Otherwise we'll have a scenario where `FAKE_GCS_PORT=abc` causes the server to start on port 443, but if we do something like `FAKE_GCS_SCHEME=foo` the server will crash, which is a bit inconsistent.

ota2000 commented:
Thanks! Honestly, I don't mind if you or anyone else wants to take over to get this merged faster. It's been sitting for a while, so feel free to pick it up if you have the time!

ota2000 commented:
@fsouza Thank you for the review! I've addressed all your feedback:

1. **Restored `t.Parallel()` in `TestToFakeGcsOptions`** — since it doesn't touch environment variables, it can safely run in parallel.
2. **Extracted `nopConverter`** — replaced all inline anonymous `func(s string) (string, error) { return s, nil }` with a named function.
3. **Propagated converter errors from `envVarOrDefault`** — changed the return type to `(T, error)` so invalid environment variable values (e.g., `FAKE_GCS_PORT=abc`) now return an error instead of silently falling back to defaults. This makes behavior consistent with flag validation.

All tests pass. If everything looks good, it would be great to get this merged. Thanks!
